### ISSUE TITLE
Openclash is not available in your Opnsense Repository

### ISSUE THREAD

**Issue by manuvaidya81** [mimugmail/opn-repo#195]
Hi, i got to setup Opnsense for my Telekom Glasfiber today. I was previously using Openwrt with Openclash as my VPN to hide my IP.
But i am not finding Openclash Plugin for Opnsense and i really need it. Shadowsocks plugin doesnt serve my purpose.

Pls guide me on how to get Openclash Plugin installed in Opnsense. Thanks in advance.

**Comment by manuvaidya81** [comment]
Ok, I understand that. Is it that difficult to build a package for opnsense? I mean freebsd and Linux both have same architecture right? If that is so difficult are there any other plugins where it works as proxy server for opnsense just like how openclash works in openwrt?

**Comment by mihakralj** [comment]
Linux and freeBSD have less in common than Linux and Android - but I will let you explore that on your own.

Writing a plugin for OPNsense is not hard - as long as plugin is enhancing something that runs on freeBSD.

What you are asking for is a system level rewrite of software that works on Linux to start working on freeBSD. Doable? Yes. But not by OPNsense enthusiasts. You need system engineering here.

**Comment by manuvaidya81** [comment]
Ok. I didn't know that. Thanks for explaining it in detail. Any other freebsd plugins you know off which works just like Openclash??

**Comment by mimugmail** [comment]
This?

https://www.freshports.org/net/clash/

**Comment by mimugmail** [comment]
https://www.reddit.com/r/OPNsenseFirewall/comments/zy5u4d/new_community_repo_plugin_clash/

**Comment by mihakralj** [comment]
Michael to the rescue! Does this need a plugin wrapper? Happy to assist with that.

**Comment by manuvaidya81** [comment]
> This?
> 
> https://www.freshports.org/net/clash/

yes. i saw it. but i dont know how i configure it. does it really work if i simply paste the config details in it after enabling it ??

**Comment by manuvaidya81** [comment]
> Michael to the rescue! Does this need a plugin wrapper? Happy to assist with that.

I installed this Plugin Clash from mimugmail repo and i enabled it. after that i tried to copy config ( yaml file) details in it but it somehow doesnt work. any idea why ??

**Comment by mimugmail** [comment]
Anything in System : Log Files : General?

**Comment by manuvaidya81** [comment]
> Anything in System : Log Files : General?

i really dont know if Clash is working and i dont know where to look for the log files specifically for Clash Plugin...

**Comment by manuvaidya81** [comment]
> Anything in System : Log Files : General?

Nothing there related to clash.... Any cmd option to find out if clash is really working or not?

**Comment by mimugmail** [comment]
/usr/local/etc/rc.d/clash restart

Maybe, I dont use it myself

**Comment by manuvaidya81** [comment]
> /usr/local/etc/rc.d/clash restart
> 
> Maybe, I dont use it myself

Where do I copy the config details ? If I try to copy the config, nothing happens. But this clash is in your repo. You should be knowing how to get it to work in opnsense properly right? Have you already tested it on opnsense if it works or not ?

**Comment by manuvaidya81** [comment]
> /usr/local/etc/rc.d/clash restart
> 
> Maybe, I dont use it myself

It says permission denied.

**Comment by mihakralj** [comment]
How about `sudo /usr/local/etc/rc.d/clash restart`

**Comment by manuvaidya81** [comment]
> How about `sudo /usr/local/etc/rc.d/clash restart`

it is saying like this....
![opnsense1](https://github.com/mimugmail/opn-repo/assets/155751439/56d237da-2d5f-4459-ae2b-24edbf848f13)
but i am still seeing the same old IP Address... that means Opnsense is not forwarding the DNS Traffic through Clash...

This is my Config in Clash.conf ..
`mixed-port: 7890
allow-lan: true
mode: Rule
log-level: info
external-controller: 0.0.0.0:9090
dns:
  enabled: true
  listen: :53530
  ipv6: true
  default-nameserver: 
    - 223.5.5.5
    - 8.8.8.8
  nameserver:
    - https://dns.alidns.com/dns-query
    - https://doh.pub/dns-query
  fallback:
    - https://1.0.0.1/dns-query
    - tls://dns.google
`
Any help pls ??

**Comment by manuvaidya81** [comment]
here is my Clash.conf contents. 
`mixed-port: 7890
allow-lan: true
mode: Rule
log-level: info
external-controller: 0.0.0.0:9090
dns:
  enabled: true
  listen: :53530
  ipv6: true
  default-nameserver: 
    - 223.5.5.5
    - 8.8.8.8
  nameserver:
    - https://dns.alidns.com/dns-query
    - https://doh.pub/dns-query
  fallback:
    - https://1.0.0.1/dns-query
    - tls://dns.google
proxies: .......

rules:
  - GEOIP,LAN,DIRECT
  - GEOIP,CN,ðŸŽ¯ å…¨çƒç›´è¿ž
  - MATCH,ðŸŸ æ¼ç½‘ä¹‹é±¼
`
Should i adjust any port/ Ip Address ?
i am using Unbound on 53530 Port and Port 53 for AdGuardhome along with DNScrypt-Proxy on 53531 Port.

i am completely lost in getting this clash to work.. i didnt have this much difficulty in setting up Openclash in Openwrt and after a bit of research and learning i got it everything finally working in it.

Should i add any rules in Firewall so that it works without problems ??
I am a complete noob in Opnsense . hence i am requesting help here... Thanks.

**Comment by mihakralj** [comment]
Is the configuration on OPNsense same as on wrt?

**Comment by manuvaidya81** [comment]
no but i altered it....should i use the same config  and try ??

**Comment by mihakralj** [comment]
We do not know. This here is not a repository of official OPNsense development - it is a repository of some enthusiasts that know just enough of freeBSD, OPNsense and coding to curate some plugin that are not part of core OPNsense.

Nobody here knows or uses clash; I never heard of it before but @mimugmail found it among freeBSD ports and now included the package to his repo here.

While we wish you success in making it work, all we can do is help YOU with troubleshooting suggestions.

**Comment by manuvaidya81** [comment]
Ok, i give up then. Thanks for your time. 
I did exactly as explained in this Post ([https://pfchina.org/?p=10526](url) but no success...
i dont know what to do....

**Comment by mimugmail** [comment]
Sorry cant help here since I only build plugins but dont use them (besides AdGuard)

**Comment by HernandoR** [comment]
I'm new to freeBSD and OPNsense. but i'm familiar with linux and wrt. I'd love to migrate Clash into OPNsense, where should i contribute?

**Comment by mimugmail** [comment]
You can install the plugin, enter your config to the text box and check if it works :)

**Comment by manuvaidya81** [comment]
> You can install the plugin, enter your config to the text box and check if it works :)

but where do i find this Openclash Plugin in your repository ? it is not available.... Would you mind sharing the steps to follow in successfully installing this Plugin and its setup on Opnsense pls ?
Thanks in advance.

**Comment by mimugmail** [comment]
System: Firmware: Plugins: search for clash .. its there
